src/pages/Sign/sections/SignUp.tsx

```diff
@@ -137,62 +196,216 @@ export const SignUp: React.SFC = () => (
+                      }
+                    >
+                      {({ errors, touched }) => (
+                        <CustomForm>
+                          <InputTitle>
+                            <TranslationsConsumer textKey="SIGN_INPUT_ONE_TITLE">
+                              {(title) => title}
+                            </TranslationsConsumer>
+                          </InputTitle>
+
+                          <InputField name="email" />
+                          {errors.email && touched.email ? (
+                            <ErrorMessage>
+                              <TranslationsConsumer textKey={errors.email}>
+                                {(errorMessage) => errorMessage}
+                              </TranslationsConsumer>
+                            </ErrorMessage>
+                          ) : null}
+                          <InputTitle>
+                            <TranslationsConsumer textKey="SIGN_INPUT_TWO_TITLE">
+                              {(title) => title}
+                            </TranslationsConsumer>
+                          </InputTitle>
+                          <InputField name="personalNumber" />
+                          {errors.personalNumber && touched.personalNumber ? (
+                            <ErrorMessage>
+                              <TranslationsConsumer
+                                textKey={errors.personalNumber}
+                              >
+                                {(errorMessage) => errorMessage}
+                              </TranslationsConsumer>
+                            </ErrorMessage>
+                          ) : null}
+                          <GetInsuredButton>
+                            <TranslationsConsumer textKey="SIGN_BUTTON_TEXT">
+                              {(buttonText) => (
+                                <InputSubmit type="submit" value={buttonText} />
+                              )}
+                            </TranslationsConsumer>
+                          </GetInsuredButton>
+                          <Subscription<SignStatusData>
+                            subscription={SIGN_SUBSCRIPTION}
+                          >
+                            {({ data, loading, error }) => {
+                              if (loading) {
+                                return null
+                              }
+                              if (error) {
+                                return (
+                                  <div>
+                                    <ErrorText>
+                                      <TranslationsConsumer textKey="SIGN_BANKID_STANDARD_ERROR_MESSAGE">
+                                        {(errorText) => errorText}
+                                      </TranslationsConsumer>
+                                    </ErrorText>
+                                  </div>
+                                )
+                              }
+                              if (data) {
```

Comment:
can we move these nested branching statements into a util functions?
a) nesting branching statements gets quite hard to read
b) We're mixing view and business logic here imo
c) I don't like branching in general, so if we could solve this in another approach i would be very happy :D